url/nodeurl.go

```diff
@@ -0,0 +1,118 @@
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type searchParam struct {
+	name  string
+	value []string
+}
+
+func (sp *searchParam) Encode() string {
+	vals := []string{}
+	for _, v := range sp.value {
+		vals = append(vals, url.QueryEscape(v))
+	}
+
+	str := url.QueryEscape(sp.name)
+	if len(vals) > 0 {
+		str = fmt.Sprintf("%s=%s", str, strings.Join(vals, ","))
+	}
+	return str
+}
+
+func (s searchParam) String() string {
```

Comment:
Oops this was an oversight. That being said I did define the ones for `searchParams` without using a pointer since it's a type alias for `[]searchParam`. I'm not sure if this is bad practice or not. But I wasn't sure how to define functions for an array type.